pkg/modulereader/resreader.go

```diff
@@ -115,11 +115,23 @@ func (i ModuleInfo) GetOutputsAsMap() map[string]OutputInfo {
 	return outputsMap
 }
 
+type sourceAndKind struct {
+	source string
+	kind   string
+}
+
+var modInfoCache = map[sourceAndKind]ModuleInfo{}
+
 // GetModuleInfo gathers information about a module at a given source using the
```

Comment:
GetModuleInfo is now a "write" operation in the sense that it caches the read operation for subsequent read operations. Please review this comment, remove anything that is no longer accurate, and describe the caching functionality.